### DISCUSSION TITLE
access_control on DAG and ImportError

### DISCUSSION THREAD

**Discussion by arjunanan6** [apache/airflow#34552] [Q&A] [answered]
I am having some troubles with a custom role definition. We use the `access_control` definition in our DAGs which map to specific roles, like: Admin-XYZ. 

This largely works, and the user with this role is only able to see DAGs with this defined in them. However, what they do not see are import errors.
The role provides the user with "can read on DAG Warnings, can read on ImportError", which I thought would be enough, but apparently not. What am I missing here?

**Comment by hussein-awala** [comment, marked as answer]
The user should see the import errors for all the dags which he can read:
https://github.com/apache/airflow/blob/ca3ce78fbaa4a04c061116acf2c86a311fdab30d/airflow/www/views.py#L933-L937

However, the dag should be created first, then broken for some reason, otherwise we cannot detect that the user can read the dag in the broken dag file.

**Reply by arjunanan6** [reply]
I just reproduced this, and indeed, it does show up if it is broken at some point, and then thereafter. However, for a first-time DAG, this is not useful, so I have just chosen to display DAG import tests in our build pipeline. Thank you for looking into this!
